hs_discover/views.py

```diff
@@ -54,10 +54,17 @@ def get(self, request, *args, **kwargs):
 
         sqs = SearchQuerySet().all()
 
+        _FACETED_FIELDS = ['subject', 'author', 'owner', 'contributor', 'creator', 'title']
         if request.GET.get('q'):
             q = request.GET.get('q')
-            sqs = sqs.filter(content=q)
-
+            if q.startswith('"') and q.endswith('"'):
+                q = q.strip('"')
+                # do exact match on all of the faceted fields when the search term is enclosed in double quotes
+                # e;g., "model my watershed"
+                for fc_field in _FACETED_FIELDS:
```

Comment:
This is a little strange. All text is indexed in the magic search field "text". All you need is one match, to "text". Matching the faceted fields is overkill. If matching to 'text' doesn't work, we need to update the template for the text field 
(~/hs_core/templates/search/indexes/hs_core/baseresource_text.txt) and then it will work properly.

Reply:
Are you depending upon the faceted fields to strip - from the terms? Not a good idea. Easier to do it manually.

Reply:
I tried exact match using the magic 'text' field and it didn't work.